nautobot/extras/models/groups.py

```diff
@@ -1,10 +1,12 @@
 """Dynamic Groups Models."""
 
 import logging
+import pickle
```

Comment:
We've generally been trying to avoid using `pickle` due to its known security concerns. I'm guessing it's necessary here since we're wanting to store and return a fully evaluated QuerySet rather than something like a simple list of PKs? Worth considering though.